packages/block-editor/src/store/effects.js

```diff
@@ -116,24 +122,26 @@ export default {
 			blocksWithTheSameType[ 0 ].attributes
 		);
 
-		const newAttributeKey = findKey( updatedAttributes, ( v ) =>
-			typeof v === 'string' && v.indexOf( START_OF_SELECTED_AREA ) !== -1
-		);
-		const convertedHtml = updatedAttributes[ newAttributeKey ];
-		const multilineTagA = blockAType.attributes[ newAttributeKey ].multiline;
-		const convertedValue = create( { html: convertedHtml, multilineTag: multilineTagA } );
-		const newOffset = convertedValue.text.indexOf( START_OF_SELECTED_AREA );
-		const newValue = remove( convertedValue, newOffset, newOffset + 1 );
-		const newHtml = toHTMLString( { value: newValue, multilineTag: multilineTagA } );
-
-		updatedAttributes[ newAttributeKey ] = newHtml;
-
-		dispatch( selectionChange(
-			blockA.clientId,
-			newAttributeKey,
-			newOffset,
-			newOffset
-		) );
+		if ( hasSelection ) {
+			const newAttributeKey = findKey( updatedAttributes, ( v ) =>
+				typeof v === 'string' && v.indexOf( START_OF_SELECTED_AREA ) !== -1
+			);
+			const convertedHtml = updatedAttributes[ newAttributeKey ];
+			const multilineTag = blockAType.attributes[ newAttributeKey ].multiline;
+			const convertedValue = create( { html: convertedHtml, multilineTag } );
+			const newOffset = convertedValue.text.indexOf( START_OF_SELECTED_AREA );
+			const newValue = remove( convertedValue, newOffset, newOffset + 1 );
+			const newHtml = toHTMLString( { value: newValue, multilineTag } );
+
+			updatedAttributes[ newAttributeKey ] = newHtml;
+
+			dispatch( selectionChange(
+				blockA.clientId,
```

Comment:
Am I correct in thinking: While the selection may be within one or the other (or neither) of the two blocks, the merge will result in the first block becoming selected (if a selection had existed previously).

Reply:
Should it? I didn't think much about it. If there was no selection previously, there's no selection to be merged?